traitlets/config/application.py

```diff
@@ -422,7 +422,7 @@ def print_help(self, classes=False):
         self.print_options()
 
         if classes:
-            help_classes = self._classes_with_config_traits()
+            help_classes = self.classes
```

Comment:
We should respect the classes present in `Application.classes`, but we cannot hide classes that are relevant when applying config-values.  
Actually, `_classes_with_config_traits()` reads this list, and augments it with any relevant baseclasses.